### PR TITLE
Show warning about localStorage size in About-IITC

### LINKED ISSUE #513: Plugins are not listed in About-IITC Dialog
If the main script is loaded first it may execute it's boot-routine before other scripts have finished loading.
This leads to an incomplete `bootPlugins.info` array (some items are still left in the `bootPlugins` array).

Tested on FF with TM, installed IITC-main first, then a ~12 other plugins.
On first start only one plugin was listed in the about dialog. On a second run it lists 6.
works fine if main script is loaded last

NOTE: since `bootPlugins` is an array `bootPlugins.info` should be moved to another variable

### PR DESCRIPTION
**Pull request by modos189**
The original [PR](https://github.com/IITC-CE/ingress-intel-total-conversion/pull/514) from [McBen](https://github.com/McBen) did not remove aboutIITC from the original file. The PR has also been updated to the new master branch

Some iitc-plugins are messing around with localStorage and may lead to out-of-quota errors. For the user this will looks like IITC is forgetting settings.

This PR will check if localStorage can take another 100kb of data. If not it will display a warning message in the about box.
grafik

The first commits are just about refactoring the old code.
[9cb5d02](https://github.com/IITC-CE/ingress-intel-total-conversion/commit/1ec6398c204f850ca0b7e6aecc8b286f81aa6422) will add the warning message.
[635a77b](https://github.com/IITC-CE/ingress-intel-total-conversion/commit/1195060cde0d97e192c7505f2cb51b85ca177352) will fix a bug discovered while working on this. 

closes https://github.com/IITC-CE/ingress-intel-total-conversion/pull/514
closes https://github.com/IITC-CE/ingress-intel-total-conversion/issues/513

### CODE REVIEW

File: core/code/dialog_about.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+function createDialogContent() {
+  var html = ''
+    + '<div><b>About IITC</b></div> '
+    + '<div>Ingress Intel Total Conversion</div> '
+    + '<hr>'
+    + '<div>'
+    + '  <a href="'+'@url_homepage@'+'" target="_blank">IITC Homepage</a> |' +
+    '  <a href="'+'@url_tg@'+'" target="_blank">Telegram channel</a><br />'
```
Comment (le-jeu) [inline core/code/dialog_about.js]: alignment

File: core/code/dialog_about.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    name: p.name,
+    verinfo: formatVerInfo(p, extra)
+  };
+
+  if (isStandardPlugin(p)) {
+    info.style += 'color:darkgray;';
+  }
+
+  if (p.error) {
+    info.style += 'text-decoration:line-through;';
```
Comment (le-jeu) [inline core/code/dialog_about.js]: maybe css classes to avoid putting style into the code ?

File: core/code/dialog_about.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  return '';
+}
+
+
+function isShortOnLocalStorage() {
+  var MINIMUM_FREE_SPACE = 100000;
+  try {
+    localStorage.setItem('_MEM_CHECK_', '#'.repeat(MINIMUM_FREE_SPACE));
+  } catch (e) {
+    console.error('out of localstorage space', e);
```
Comment (le-jeu) [inline core/code/dialog_about.js]: not using `ulog` ?

File: core/code/dialog_about.js (hunk 1 above)
Comment (modos189) [reply core/code/dialog_about.js]: done

File: core/code/dialog_about.js (hunk 2 above)
Comment (modos189) [reply core/code/dialog_about.js]: done

File: core/code/dialog_about.js (hunk 3 above)
Comment (modos189) [reply core/code/dialog_about.js]: done

**Pull request merged**